File: android/app/src/main/java/com/fmaldonado/siase/data/repositories/KardexRepository.kt
package com.fmaldonado.siase.data.repositories

import com.fmaldonado.siase.data.models.Careers
import com.fmaldonado.siase.data.models.Kardex
import com.fmaldonado.siase.data.network.NetworkDataSource
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class KardexRepository
@Inject
constructor(
    private val authRepository: AuthRepository,
    private val networkDataSource: NetworkDataSource,
) {

    suspend fun getKardex(career: Careers): Kardex {
        val index = authRepository.signedInUser!!.carreras.indexOfFirst {
            it.claveCarrera == career.claveCarrera &&
                    it.claveDependencia == career.claveDependencia
        }

        return networkDataSource.getKardex(index)
    }
}

File: ios/siase/data/repositories/KardexRepository.swift
//
//  KardexRepository.swift
//  siase
//
//  Created by Fernando Maldonado on 22/04/22.
//

import Foundation

class KardexRepository{
    
    private let authRepository:AuthRepository
    private let networkDataSource:NetworkDataSource
    
    init(
        authRepository:AuthRepository = DIContainer.shared.resolve(type: AuthRepository.self)!,
        networkDataSource:NetworkDataSource = DIContainer.shared.resolve(type: NetworkDataSource.self)!
    ){
        self.authRepository = authRepository
        self.networkDataSource = networkDataSource
    }
    
    func getKardex(career:Career){
        
    }
    
}
